### ISSUE TITLE
Support slots

### ISSUE THREAD

**Issue by dantownsend** [pallets/jinja#1651]
First off, apologies if this already exists and I'm not aware.

In other templating libraries you have the concept of slots ([Vue JS](https://v2.vuejs.org/v2/guide/components-slots.html#Slot-Content) is perhaps the most famous example).

Imagine this scenario where we want to populate a modal with some content:

```html
{% with content="<p>This is the inner content</p>" %}
    {% include "modal.html.jinja" %}
{% endwith %}
```

The `modal.html.jinja` file would look something like:

```html
<div class="modal">
    {{ content|safe }}
</div>
```

It soon gets unwieldy if the `content` value is very large.

Using the proposed feature, we would be able to do this instead:

```html
{% include "modal.html.jinja" %}<p>This is the inner content</p>{% endinclude %}
```

And then `modal.html.jinja` would be this insted:

```html
<div class="modal">
    {% slot %}
</div>
```

In summary, a way to pass large HTML fragments to an include statement. Thanks!

**Comment by davidism** [comment]
You're looking for macros: https://jinja.palletsprojects.com/en/3.1.x/templates/#macros
